components/pd_client/src/util.rs

```diff
@@ -220,17 +223,18 @@ impl Client {
     ///
     /// If `force` is false, it will reconnect only when members change.
     pub async fn reconnect(&self, force: bool) -> Result<()> {
-        let (future, start) = {
+        let start = Instant::now();
+
+        let future = {
             let inner = self.inner.rl();
-            if inner.last_update.elapsed() < Duration::from_secs(RECONNECT_INTERVAL_SEC) {
+            if start.duration_since(inner.last_try_reconnect) < GLOBAL_RECONNECT_INTERVAL {
```

Comment:
`start` is initialized without lock, so it can be less than `last_try_reconnect`. Should use `checked_duration_since` instead.